open-ce/utils.py

```diff
@@ -150,3 +152,39 @@ def get_output(command):
     '''Print and execute a shell command and then return the output.'''
     print("--->{}".format(command))
     return subprocess.check_output(command, shell=True).decode("utf-8").strip()
+
+def variant_key(py_ver, build_type):
+    '''
+    Returns a variant key using python version and build type
+    '''
+    result = ""
+    if py_ver:
+        result +=  "py" + py_ver
+    if build_type:
+        result +=  "-" + build_type
+    return result
+
+def generalize_version(package):
+    """Add `.*` to package versions when it is needed."""
+
+    # Remove multiple spaces or tabs
+    package = re.sub(' +', ' ', package)
+
+    # Check if we want to add .* to the end of versions
+    py_matched = re.match(r'(\w+[-]*\w+)([\s,=,<,>]*)(.*)', package)
+
+    if py_matched:
+        name = py_matched.group(1)
+        operator = py_matched.group(2)
+        version = py_matched.group(3)
+        append_asterik = False
+
+        if len(version) > 0 and len(operator) > 0:
+
+            #Append .* at the end if it is not there and if operator is space or == or = or empty
+            if not version.endswith(".*") and operator.strip() in ["=", "==", " ", ""]:
```

Comment:
If you're doing `operator.strip()`, I don't think you need both `" "` and `""`.
Also, I don't think we need to add th `.*` with just a single `=`.

Reply:
"" was needed when operator contains all the spaces. In that case strip() removes all the spaces returns empty string. 
For `=`, yes we can remove that. However, I'd kept it for consistency since we are adding .* for ==, although for =, .* isn't needed.